File: eBarberShop.Services/Database/Rezervacija.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eBarberShop.Services.Database
{
    public class Rezervacija
    {
        public int RezervacijaID { get; set; }
        public int KorisnikID { get; set; }
        public int? TerminID { get; set; }
        public int UslugaID { get; set; }
        public DateTime DatumRezervacije { get; set; }
        public bool IsCanceled { get; set; } = false;
        public bool IsArchived { get; set; } = false;
        public virtual Korisnik Korisnik { get; set; } = null!;
        public virtual Termin Termin { get; set; } = null!;
        public virtual Usluga Usluga { get; set; } = null!;
    }
}

File: ebarbershop_mobile/lib/models/rezervacija.dart
import 'package:ebarbershop_mobile/models/korisnik.dart';
import 'package:ebarbershop_mobile/models/termin.dart';
import 'package:ebarbershop_mobile/models/usluga.dart';
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija.g.dart';

@JsonSerializable()
class Rezervacija{
  int? rezervacijaID;
  int? korisnikID;
  int? terminID;
  int? uslugaID;
  DateTime? datumRezervacije;
  bool? isCanceled;
  bool? isArchived;
  Korisnik? korisnik;
  Termin? termin;
  Usluga? usluga;

  Rezervacija();

  factory Rezervacija.fromJson(Map<String, dynamic> json) => _$RezervacijaFromJson(json);

    Map<String, dynamic> toJson() => _$RezervacijaToJson(this);
}